File: БР1.1/Захматов Юрий/labs/lab2/reviews-service/src/main/java/com/petproject/reviewsservice/entities/ReviewEntity.java
package com.petproject.reviewsservice.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Table(name = "review")
@Entity
@NoArgsConstructor
@Getter
@Setter
@Builder
@AllArgsConstructor
public class ReviewEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @Column(name="user_id", nullable = false)
    private Long userId;

    @Column(name="property_id", nullable = false)
    private Long propertyId;

    @Column(nullable = false)
    private Integer rating;

    private String comment;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = this.createdAt;
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: БР1.2/Москалец Данила/labs/lab2/sangel-api/order-service/src/entities/review.entity.ts
import { Entity, Column, ManyToOne, JoinColumn, Unique } from 'typeorm';
import { BaseEntity } from '../common/base.entity';

@Entity({ name: 'reviews' })
@Unique(['service_id', 'user_id'])
export class Review extends BaseEntity {
  @Column({ name: 'service_id' })
  service_id: number;

  @Column({ name: 'user_id' })
  user_id: number;

  @Column({ type: 'integer' })
  rating: number;

  @Column({ type: 'text', nullable: true })
  comment: string | null;
}